### ISSUE TITLE
Setting "ServiceInstance" through "CustomNavSettings"-env?

### ISSUE THREAD

**Issue by CloudViking86** [microsoft/nav-docker#430]
Environment variable used;
`-e CustomNavSettings=ServerInstance=MyNavInstance`

Bear with me and I'll explain how I think it works;

I set the environment variable above and it edits the KEY-VALUE pair in "CustomSettings.config" in the containers "C:\Program Files\Microsoft Dynamics NAV\<version>\Service"-folder.

If I look at the description of all the scripts and what they do here;
https://github.com/microsoft/nav-docker/blob/master/HOWTO.md
I can see that the initial script "c:\run\navstart.ps1" creates the "$serverInstance"-variable through these lines;

```
$serviceTierFolder = (Get-Item "C:\Program Files\Microsoft Dynamics NAV\*\Service").FullName
$CustomConfigFile = Join-Path $serviceTierFolder "CustomSettings.config"
$CustomConfig = [xml](Get-Content $CustomConfigFile)
$serverInstance =$CustomConfig.SelectSingleNode("//appSettings/add[@key='ServerInstance']").Value
```

If I run these commands in my container and echo out "$serverInstance" or check "CustomSettings.config" I get "MyNavInstance" as the "ServerInstance" however when importing the NavAdminTool I get that the service is named the default;
```
ServerInstance : MicrosoftDynamicsNavServer$NAV                                                                   
DisplayName    : Microsoft Dynamics NAV Server [NAV]                                                              
State          : Running                                                                                              
ServiceAccount : NT AUTHORITY\SYSTEM                                                                                    
Version        : 10.0.15140.0                                                                                          
Default        : True 
```

Now there is a file called "ServiceSettings.ps1" in "C:\Run" which by default contains this;
```
$NavServiceName = 'MicrosoftDynamicsNavServer$NAV'
$WebServerInstance = "NAV"
$ServerInstance = "NAV"
```

I can see in "SetupVariables.ps1" here;
https://github.com/microsoft/nav-docker/blob/master/generic/Run/SetupVariables.ps1
that there is an environment variable called "$WebServerInstance" but there does not seem to be one for "$NavServiceName" or "$ServerInstance".

TL;DR:
I want to set the name of my Nav-service to a custom value WITHOUT having to overwrite any scripts, I could always just create my own hard-coded "ServiceSettings.ps1" but I would like to have it dynamic.

As always, appreciate your time and assistance - CloudViking86

**Comment by freddydk** [comment]
$ServerInstance is set in ServiceSettings - NAV for 14.x and before, BC for 15.x and later.
You might run into a lot of issues if you change this or make it dynamic.

**Comment by CloudViking86** [comment]
> $ServerInstance is set in ServiceSettings - NAV for 14.x and before, BC for 15.x and later.
> You might run into a lot of issues if you change this or make it dynamic, and you cannot reliably change it in other scripts.

Thanks for your very fast reply as usual :)

Ok so even if I set;
`-e CustomNavSettings=ServerInstance=MyNavInstance` that doesn't do anything since it is controlled by the "ServiceSettings.ps1" and changing that would involve overwriting that script since passing an environment variable such as "$WebServerInstance" which I saw you added in commit 2f5bf1148a38f7f6893eea505951078140739022 here isn't available (for good reason since as you said it might cause a lot of issues down the line)?

**Comment by freddydk** [comment]
Correct, then you will just end up with a container that doesn't work I guess

**Comment by CloudViking86** [comment]
Cheers Freddy
